features/upgrade/routing/upgrade.feature

```diff
@@ -208,3 +208,82 @@ Feature: Routing and DNS related scenarios
     Then the step should succeed
     And the output should contain "Hello-OpenShift web-server-rc"
     """
+
+  @upgrade-prepare
+  @users=upuser1,upuser2
+  Scenario: Unidling a route work without user intervention - prepare
+    Given I switch to first user
+    And I run the :new_project client command with:
+      | project_name | ocp45955 |
+    Then the step should succeed
+    When I use the "ocp45955" project
+    Given I obtain test data file "routing/web-server-rc.yaml"
+    When I run the :create client command with:
+      | f | web-server-rc.yaml|
+    Then the step should succeed
+
+    And a pod becomes ready with labels:
+      | name=web-server-rc |
+    Then the expression should be true> service('service-unsecure').exists?
+    When I expose the "service-unsecure" service
+    Then the step should succeed
+
+    Given I wait until number of replicas match "1" for replicationController "web-server-rc"
+    Given I have a test-client-pod in the project
+    When I execute on the pod:
+      | curl |
+      | -ksS |
+      | http://<%= route("service-unsecure", service("service-unsecure")).dns(by: user) %>/ |
+    Then the step should succeed
+    And the output should contain "Hello-OpenShift web-server-rc"
+
+    When I run the :idle client command with:
+      | svc_name | service-unsecure |
+    Then the step should succeed
+    Given I wait until number of replicas match "0" for replicationController "web-server-rc"
+
+    # Check the servcie service-unsecure to see the idle annotation
+    When I run the :get client command with:
+      | resource      | service          |
+      | resource_name | service-unsecure |
+      | n             | ocp45955         |
+      | o             | yaml             |
+    Then the step should succeed
+    And the output should contain:
+      | idling.alpha.openshift.io/unidle-targets: '[{"kind":"ReplicationController","name":"web-server-rc","replicas":1}]' |
```

Comment:
Above lines can be shorten as,
`And the expression should be true> service('service-unsecure').annotation('idling.alpha.openshift.io/unidle-targets') == "xxxx"`

Reply:
done

Reply:
@liangxia We tried many times seems the `the expression should be true> service('service-unsecure')` always get the stale data that before the annotation is added. any idea?

the output of debugging
```
[1] pry(#<BushSlicer::DefaultWorld>)> service('service-unsecure')
          "operation"=>"Update",
          "time"=>"2021-12-01T10:33:33Z"}],
       "name"=>"service-unsecure",
       "namespace"=>"ocp45955",
       "resourceVersion"=>"236684",
       "uid"=>"703aeab7-23ab-46c1-8016-77655489a33d"},
```
the output from `get client command`
```
          operation: Update
          time: "2021-12-01T10:33:53Z"
        name: service-unsecure
        namespace: ocp45955
        resourceVersion: "236841"
        uid: 703aeab7-23ab-46c1-8016-77655489a33d
```

Reply:
@lihongan  Add `cached: false` to force a refresh.

Reply:
service('service-unsecure').annotation('idling.alpha.openshift.io/unidle-targets', cached: false) == "xxxx"

Reply:
awesome, works like a charm. Thank you!

Reply:
@liangxia 
Changes done and it is working
Thanks